components/clinical/timeline/src/organisms/timeline.tsx

```diff
@@ -166,8 +208,35 @@ const Timeline: FC<IProps> = ({ timelineItems, domainResourceType }) => {
 
   let position = 0
 
+  const handleFilterChange = (key: number, filter?: string) => {
+    const newActiveFilters = { ...activeFilters }
+
+    if (filter && filter.length > 0) {
+      newActiveFilters[key] = filter
+    } else {
+      delete newActiveFilters[key]
+    }
+
+    setActiveFilters(newActiveFilters)
+    onFilterChange && onFilterChange(Object.values(newActiveFilters).filter((x) => x && x.length > 0))
+  }
+
   return (
     <StyledTimeline key="timeline" data-testid="timeline">
+      {filters && (
```

Comment:
Just from playing with the example in Storybook, I wonder if it's worth making the filters static so they remain at the top of the screen as you scroll down? Especially if Care Plans are going to have enough items to require infinite scrolling

Reply:
Excellent idea 🙂